From a5d652920b61646776f244321d4321ba5164deb4 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 13 Aug 2013 20:08:42 +0000
Subject: [PATCH 0466/1077] graphic library V1.0.0.4 Mantis 0028572: GPPDev:
 Receipt support Barcode according to AEA2011 added field size_t columns to
 BarcodePDF417Settings if rows != 0 use rows if columns != 0 use columns else
 use ratio of height/width = 3/4

git-svn-id: svn://localhost/SelfServiceCommon/trunk@567 90b65887-3827-0410-9a23-83215b262276
---
 .../Interfaces/inc/graphic/barcodePDF417.h    | 12 +++-------
 .../Massai/cpp/Graphic/src/barcodePDF417.cpp  | 24 +++++++++++--------
 .../Massai/cpp/Graphic/src/local_def.cpp      | 10 +++++++-
 3 files changed, 26 insertions(+), 20 deletions(-)

diff --git a/SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h b/SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h
index 1c52df7b..974f789a 100644
--- a/SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h
+++ b/SelfServiceCommon/Interfaces/inc/graphic/barcodePDF417.h
@@ -1,12 +1,4 @@
-/**
- **************************************************************************
- *
- * @file barcodePDF417.h
- *
- * create barcodePDF417 bitmap
- *
- **************************************************************************
-**/
+// create barcodePDF417 bitmap
 #ifndef BARCODEPDF417_INCLUDED
 #define BARCODEPDF417_INCLUDED
 #ifdef SHOW_INCLUDE_FILES
@@ -27,7 +19,9 @@ namespace MASSAI
       size_t rows;
       size_t errorLevel;
       bool   useQuietZone;
+      size_t columns;
     };
+
     class BarcodePDF417: public Barcode
     {
       public:
diff --git a/SelfServiceCommon/Massai/cpp/Graphic/src/barcodePDF417.cpp b/SelfServiceCommon/Massai/cpp/Graphic/src/barcodePDF417.cpp
index aecbc63c..863ab220 100644
--- a/SelfServiceCommon/Massai/cpp/Graphic/src/barcodePDF417.cpp
+++ b/SelfServiceCommon/Massai/cpp/Graphic/src/barcodePDF417.cpp
@@ -1,13 +1,4 @@
-/**
- *  @file      pdf417BMP.cpp
- *
- *  @brief     create pdf in windows bmp fileformat (black/white)
- *
- *  @date      24.05.2006
- *
- *  @author    Materna Information & Communications (RB, SteT)
- *
- ************************************************************************/
+//  create pdf in windows bmp fileformat (black/white)
 
 #include "graphic/barcodePDF417.h"
 #include "pdf417lib.h"
@@ -54,16 +45,20 @@ namespace
           }
         }
       }
+
       ~PDF417PixelRows()
       {}
+
       size_t getHeight() const
       {
         return m_rows * m_blockHeight;
       }
+
       size_t getWidth() const
       {
         return m_width;
       }
+
       const MASSAI::text::StringBuffer& getRow(size_t y) const
       {
         if(y < getHeight())
@@ -75,6 +70,7 @@ namespace
           return m_bitmapRows[0];
         }
       }
+
     private:
       size_t m_width;
       size_t m_rows;
@@ -98,6 +94,11 @@ namespace
       params.options     |= PDF417_FIXED_ROWS;
       params.codeRows    = settings.rows;
     }
+    else if(settings.columns > 0)
+    {
+      params.options     |= PDF417_FIXED_COLUMNS;
+      params.codeColumns = settings.columns;
+    }
     else
     {
       params.options     |= PDF417_USE_ASPECT_RATIO;
@@ -129,8 +130,10 @@ namespace MASSAI
     BarcodePDF417::BarcodePDF417(const BarcodePDF417Settings& settings):
       m_settings(settings)
     {}
+
     BarcodePDF417::~BarcodePDF417()
     {}
+
     MASSAI::Bitmap::Bitmap BarcodePDF417::createBarcode(const std::string& barcodeData) const
     {
       MASSAI::Bitmap::Bitmap result;
@@ -138,6 +141,7 @@ namespace MASSAI
       BlackAndWhiteBitmap bitmap;
       return bitmap.createBitmap(pixel);
     }
+
     std::string BarcodePDF417::createBarcodeData(const std::string& data) const
     {
       return data;
diff --git a/SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp b/SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp
index d2d688d2..8c4ac14d 100644
--- a/SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp
+++ b/SelfServiceCommon/Massai/cpp/Graphic/src/local_def.cpp
@@ -1,10 +1,18 @@
 #include "Tools/VERSIONLIB.h"
 
-VERSION_ID_DT(graphic, 1, 0, 0, 3);
+VERSION_ID_DT(graphic, 1, 0, 0, 4);
 
 /*!
 \page graphic graphic library - Release Notes
 
+\date 13.08.2013
+\li V1.0.0.4
+\li Mantis 0028572: GPPDev: Receipt support Barcode according to AEA2011
+added field size_t columns to BarcodePDF417Settings
+if rows != 0 use rows
+if columns != 0 use columns
+else use ratio of height/width = 3/4
+
 \date 09.07.2013
 \li V1.0.0.3
 \li Mantis 0027971: GPPDEV: barcode 128 wrong checkdigit for Type B and Type C
-- 
2.41.0.windows.1

